src/server/components/dashboard/controllers.ts

```diff
@@ -349,3 +365,133 @@ export async function feedbackController(
     next(error);
   }
 }
+
+export async function listsEditAnnualReviewDateController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+    const { listId } = req.params;
+    const { missingValuesError, leapYearError, maxDateError } = req.query;
+    let error: Partial<QuestionError> = {};
+
+    const list = await findListById(listId);
+    const annualReviewStartDate = formatAnnualReviewDate((list as List), "annualReviewStartDate");
+
+    if (missingValuesError) {
+      error = {
+        text: "Enter a date for the annual review",
+        href: "#annualReviewDateForm",
+      };
+    }
+
+    if (leapYearError) {
+      error = {
+        text: "You cannot set the annual review to this date. Please choose another",
+        href: "#annualReviewDateForm",
+      };
+    }
+
+    if (maxDateError) {
+      error = {
+        text: "You can only change the date up to 6 months after the current review date",
+        href: "#annualReviewDateForm",
+      };
+    }
+
+    res.render("dashboard/lists-edit-annual-review-date", {
+      ...DEFAULT_VIEW_PROPS,
+      error,
+      annualReviewStartDate,
+      list,
+      csrfToken: getCSRFToken(req),
+    });
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDateController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+export async function listsEditAnnualReviewDatePostController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+
+    return (req.body.action === "confirmNewDate")
+      ? await confirmNewAnnualReviewDate(req, res)
+      : await updateNewAnnualReviewDate(req, res);
+
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDatePostController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+async function confirmNewAnnualReviewDate(req: Request, res: Response): Promise<void> {
+  const todaysDate = new Date();
+
+  const { listId } = req.params;
+  const list = await findListById(listId);
+  let annualReviewYear = todaysDate.getFullYear() + 1;
```

Comment:
This is a good point. Equally if the date is 25/12/2022 and you set it to 1/20 you'd expect that to be next year and not the current year. I think IMO it should be the same year as the next scheduled annual review. Would that make more sense?

Reply:
so if I understand correctly we're trying to 

1. determine which year the user may be referring to (this or next year)
2. validate the date is valid (not a leap year etc)
3. validate the date the user requested is within 6 months
4. determine the Date(time) of the annual review date. 


date-fns should be able to help you here. I think you will need a mixture of 
- add (to get maxDate)
- isInFuture 
- isBefore
- isValid

To resolve the "which year are they referring to" problem, do something like 

```

const possibleDates = [
  new Date(today.getFullYear, month - 1, day),
  new Date(today.getFullYear + 1, month - 1, day)
]

const futureDates = possibleDates.filter(isInFuture)
const requestedDate = futureDates.find(date => isBefore(date, maxDate));

```

Reply:
I'm a bit hesitant to jump the gun on this one because before I started the ticket I asked Neil if it would make sense to add a year input for the user, he said this would need some more discussion. So I'll have a quick chat about it with @alisalaman-cyb and Neil before we decide what to do. But this approach makes sense.

Reply:
adding a year field is ultimately unnecessary.. and opens up the door to user error in my humble onion 🧅

Reply:
I've made some changes so that if a user sets a valid date in the future the year automatically changes. Let me know your thoughts.

Reply:
I think Jen's style is probably more readable but a bit of tomARtoe / tomAYtoe, nothing to hold up the changes for.